modules/raft/src/integrationTest/java/org/apache/ignite/raft/server/ItJraftCounterServerTest.java

```diff
@@ -709,6 +733,85 @@ public void testTimerThreadsCount() {
                 "All timer threads: " + timerThreads.toString());
     }
 
+    /**
+     * The test shows that all committed updates are applied after a RAFT group restart automatically.
+     * Actual data be available to read from state storage (not a state machine) directly just after the RAFT node started.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testApplyUpdatesOutOfSnapshot() throws Exception {
+        HashMap<Integer, AtomicInteger> counters = new HashMap<>(3);
```

Comment:
Why do you declare variables like `HashMap` and not `Map`? This looks very out of place

Reply:
Because I use the implementation here.
Using interfaces is helpfully for the case when you receive an implementation from other code.

Reply:
I don't understand this reply. Do you use specific methods of HashMap that don't exist in Map?